src/amo/components/AddonReview.js

```diff
@@ -0,0 +1,135 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import { asyncConnect } from 'redux-connect';
+
+import { submitReview } from 'amo/api';
+import { setReview } from 'amo/actions/reviews';
+import { callApi } from 'core/api';
+import translate from 'core/i18n/translate';
+
+import 'amo/css/AddonReview.scss';
+
+
+export class AddonReviewBase extends React.Component {
+  static propTypes = {
+    apiState: PropTypes.object,
+    i18n: PropTypes.object.isRequired,
+    review: PropTypes.object.isRequired,
+    updateReviewText: PropTypes.func.isRequired,
+  }
+
+  static contextTypes = {
+    router: PropTypes.object.isRequired,
+  }
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    const body = this.reviewTextarea.value;
+    if (!body) {
+      // TODO: dispatch an error action when we have support for that.
+      throw new Error('The review text cannot be empty');
+    }
+    const params = {
+      body,
+      addonId: this.props.review.addonSlug,
+      reviewId: this.props.review.id,
+      apiState: this.props.apiState,
+      router: this.context.router,
+    };
+    // TODO: render a progress indicator in the UI.
+    this.props.updateReviewText(params);
+  }
+
+  render() {
+    const { i18n, review } = this.props;
+    if (!(review && review.id && review.addonSlug)) {
```

Comment:
won't that trigger an error if review is null? You can't try to access a property on null.